signature/src/error.rs

```diff
@@ -0,0 +1,67 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub enum Error {}
```

Comment:
@Ryanmtate We most likely will, I just wanted to handle that as part of #23 so that we can keep the ball moving. This PR changes a lot so I wanted to close it as quick as possible